src/solvers/dgsem_tree/indicators_2d.jl

```diff
@@ -125,6 +125,292 @@ function (indicator_hg::IndicatorHennemannGassner)(u::AbstractArray{<:Any,4},
+      u_local = get_node_vars(u, equations, dg, i, j, element)
+      indicator[i, j] = indicator_hg.variable(u_local, equations)
+    end
+
+    # Convert to modal representation
+    multiply_scalar_dimensionwise!(modal, dg.basis.inverse_vandermonde_legendre, indicator, modal_tmp1)
+
+    # Calculate total energies for all modes, without highest, without two highest
+    total_energy = zero(eltype(modal))
+    for j in 1:nnodes(dg), i in 1:nnodes(dg)
+      total_energy += modal[i, j]^2
+    end
+    total_energy_clip1 = zero(eltype(modal))
+    for j in 1:(nnodes(dg)-1), i in 1:(nnodes(dg)-1)
+      total_energy_clip1 += modal[i, j]^2
+    end
+    total_energy_clip2 = zero(eltype(modal))
+    for j in 1:(nnodes(dg)-2), i in 1:(nnodes(dg)-2)
+      total_energy_clip2 += modal[i, j]^2
+    end
+
+    # Calculate energy in lower modes
+    energy = max((total_energy - total_energy_clip1) / total_energy,
+                 (total_energy_clip1 - total_energy_clip2) / total_energy_clip1)
+
+    alpha_element = 1 / (1 + exp(-parameter_s / threshold * (energy - threshold)))
+
+    # Take care of the case close to pure DG
+    if alpha_element < alpha_min
+      alpha_element = zero(alpha_element)
+    end
+
+    # Take care of the case close to pure FV
+    if alpha_element > 1 - alpha_min
+      alpha_element = one(alpha_element)
+    end
+
+    # Clip the maximum amount of FV allowed
+    alpha[element] = min(alpha_max, alpha_element)
+  end
+
+  if (alpha_smooth)
+    # Diffuse alpha values by setting each alpha to at least 50% of neighboring elements' alpha
+    # Copy alpha values such that smoothing is indpedenent of the element access order
+    alpha_tmp .= alpha
+
+    # Loop over elements, because there is no interface container
+    for element in eachelement(dg,cache)
+      # Get neighboring element ids
+      left  = cache.elements.left_neighbors[1, element]
+      right = cache.elements.right_neighbors[1, element]
+      up    = cache.elements.left_neighbors[2, element] 
+      down  = cache.elements.right_neighbors[2, element]
+
+      # Apply smoothing
+      alpha[left]     = max(alpha_tmp[left],    0.5 * alpha_tmp[element], alpha[left])
+      alpha[element]  = max(alpha_tmp[element], 0.5 * alpha_tmp[left],    alpha[element])
+      
+      alpha[right]    = max(alpha_tmp[right],   0.5 * alpha_tmp[element], alpha[right])
+      alpha[element]  = max(alpha_tmp[element], 0.5 * alpha_tmp[right],   alpha[element])
+
+      alpha[up]       = max(alpha_tmp[up],      0.5 * alpha_tmp[element], alpha[up])
+      alpha[element]  = max(alpha_tmp[element], 0.5 * alpha_tmp[up],      alpha[element])
+
+      alpha[down]     = max(alpha_tmp[down],    0.5 * alpha_tmp[element], alpha[down])
+      alpha[element]  = max(alpha_tmp[element], 0.5 * alpha_tmp[down],    alpha[element])
+      
+    end
+
+  end
+
+  return alpha 
+end
+
+
+function (indicator_hg::IndicatorHennemannGassner)(u::AbstractArray{<:Any,4}, mesh::P4estMesh{2},
```

Comment:
Same here

Reply:
The `P4estMesh` data structures used here are exactly identical to the `TreeMesh` ones, except that someone (me) used the name `element_ids` instead of `neighbor_ids` for some reason.
Renaming `element_ids` to `neighbor_ids` in the `P4estMesh` data structures (interfaces and mortars) allows to reuse the `TreeMesh` implementation of the indicator without any change.

Reply:
Yes, thats the exact reason. I will fix this.

Reply:
fixed in 3bad0d4.